lib/Runtime/Debug/DebugContext.cpp

```diff
@@ -170,14 +189,10 @@ namespace Js
             return hr;
         }
 
-        // Cache ScriptContext as multiple calls below can go out of engine and ScriptContext can be closed which will delete DebugContext
-        Js::ScriptContext* cachedScriptContext = this->scriptContext;
-
         utf8SourceInfoList->MapUntil([&](int index, Js::Utf8SourceInfo * sourceInfo) -> bool
         {
-            if (cachedScriptContext->IsClosed())
+            if (this->IsSelfOrScriptContextClosed())
```

Comment:
No, that's the crux of this change. If during host call ScriptContext is closed we would have closed debug context but not deleted it so debugcontext will always be valid and in closed state.